### PR TITLE
[docs] IP-Adapter image embedding

### PR DESCRIPTION
**Pull request by stevhliu**
Continuation of #7016 with some light edits to the `ip_adapter_image_embeds` and masking sections, and adds `IPAdapterMaskProcessor.preprocess` to the API docs.

### CODE REVIEW

**Review by yiyixuxu — APPROVED** [review]
thanks!

File: docs/source/en/using-diffusers/ip_adapter.md [hunk 1]
```diff
@@ -25,6 +25,9 @@ Let's take a look at how to use IP-Adapter's image prompting capabilities with t
 
 In all the following examples, you'll see the [`~loaders.IPAdapterMixin.set_ip_adapter_scale`] method. This method controls the amount of text or image conditioning to apply to the model. A value of `1.0` means the model is only conditioned on the image prompt. Lowering this value encourages the model to produce more diverse images, but they may not be as aligned with the image prompt. Typically, a value of `0.5` achieves a good balance between the two prompt types and produces good results.
 
+> [!TIP]
+> In the examples below, try adding `low_cpu_mem_usage=True` to the [`~loaders.IPAdapterMixin.load_ip_adapter`] method to speed up the loading time by only loading the pretrained weights and not initializing them.
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/ip_adapter.md:29]:
```suggestion
> In the examples below, try adding `low_cpu_mem_usage=True` to the [`~loaders.IPAdapterMixin.load_ip_adapter`] method to speed up the loading time by only loading the pretrained weights.
```

File: docs/source/en/using-diffusers/ip_adapter.md [hunk 2]
```diff
@@ -231,10 +234,21 @@ export_to_gif(frames, "gummy_bear.gif")
 </hfoption>
 </hfoptions>
 
+## Configure parameters
+
+There are a couple IP-Adapter parameters that are useful to know about and can help you with your image generation tasks. These parameters can make your workflow more efficient or give you more control over image generation.
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/ip_adapter.md:239]:
```suggestion
There are a couple of IP-Adapter parameters that are useful to know about and can help you with your image generation tasks. These parameters can make your workflow more efficient or give you more control over image generation.
```

I know both "couple parameters" and "couple of parameters" are widely used but I guess the latter is slightly more preferable in the educational context?

File: docs/source/en/using-diffusers/ip_adapter.md [hunk 3]
```diff
@@ -264,8 +275,77 @@ images = pipeline(
...
+
+Binary masks specify which portion of the output image should be assigned to an IP-Adapter. This is useful for composing more than one IP-Adapter image. For each input IP-Adapter image, you must provide a binary mask an an IP-Adapter.
+
+To start, preprocess the input IP-Adapter images with the [`~image_processor.IPAdapterMaskProcessor.preprocess()`] to generate their masks. For optimal results, provide the output height and width to [`~image_processor.IPAdapterMaskProcessor.preprocess()`]. This ensures masks with different aspect ratios are appropriately stretched. If the input masks already match the aspect ratio of the generated image, you don't have to set the `height` and `width`.
+
+```py
+from diffusers.image_processor import IPAdapterMaskProcessor
+
+mask1 = load_image("https://huggingface.co/datasets/YiYiXu/testing-images/resolve/main/ip_mask_mask1.png")
+mask2 = load_image("https://huggingface.co/datasets/YiYiXu/testing-images/resolve/main/ip_mask_mask2.png")
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/ip_adapter.md:288]: We should move these to the `diffusers` docs repository.

File: docs/source/en/api/loaders/ip_adapter.md [hunk 4]
```diff
@@ -23,3 +23,7 @@ Learn how to load an IP-Adapter checkpoint and image in the IP-Adapter [loading]
 ## IPAdapterMixin
 
 [[autodoc]] loaders.ip_adapter.IPAdapterMixin
+
+## IPAdapterMaskProcessor
+
+[[autodoc]] image_processor.IPAdapterMaskProcessor
```
Comment (sayakpaul) [inline docs/source/en/api/loaders/ip_adapter.md:29]: Do we have docstrings for this already?

File: docs/source/en/using-diffusers/ip_adapter.md [hunk 5]
```diff
@@ -264,8 +275,77 @@ images = pipeline(
...
+    <img class="rounded-xl" src="https://huggingface.co/datasets/YiYiXu/testing-images/resolve/main/ip_mask_mask2.png"/>
+    <figcaption class="mt-2 text-center text-sm text-gray-500">mask two</figcaption>
+  </div>
+</div>
+
+When there is more than one input IP-Adapter image, load them as a list to ensure each image is assigned to a different IP-Adapter.
+
+```py
+face_image1 = load_image("https://huggingface.co/datasets/YiYiXu/testing-images/resolve/main/ip_mask_girl1.png")
+face_image2 = load_image("https://huggingface.co/datasets/YiYiXu/testing-images/resolve/main/ip_mask_girl2.png")
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/ip_adapter.md:312]: Should the masks correspond to these images? Maybe clarify that?

File: docs/source/en/using-diffusers/ip_adapter.md [hunk 6]
```diff
@@ -264,8 +275,77 @@ images = pipeline(
...
+    <img class="rounded-xl" src="https://huggingface.co/datasets/YiYiXu/testing-images/resolve/main/ip_mask_mask1.png"/>
+    <figcaption class="mt-2 text-center text-sm text-gray-500">mask one</figcaption>
+  </div>
+  <div class="flex-1">
+    <img class="rounded-xl" src="https://huggingface.co/datasets/YiYiXu/testing-images/resolve/main/ip_mask_mask2.png"/>
+    <figcaption class="mt-2 text-center text-sm text-gray-500">mask two</figcaption>
+  </div>
+</div>
+
+When there is more than one input IP-Adapter image, load them as a list to ensure each image is assigned to a different IP-Adapter.
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/ip_adapter.md:308]: @yiyixuxu just to confirm can we assign multiple images to a single IP Adapter? If so, I think this line needs to rephrased.

File: docs/source/en/using-diffusers/ip_adapter.md [hunk 7]
```diff
@@ -264,8 +275,77 @@ images = pipeline(
...
+    <img class="rounded-xl" src="https://huggingface.co/datasets/YiYiXu/testing-images/resolve/main/ip_mask_girl1.png"/>
+    <figcaption class="mt-2 text-center text-sm text-gray-500">IP-Adapter image one</figcaption>
+  </div>
+  <div class="flex-1">
+    <img class="rounded-xl" src="https://huggingface.co/datasets/YiYiXu/testing-images/resolve/main/ip_mask_girl2.png"/>
+    <figcaption class="mt-2 text-center text-sm text-gray-500">IP-Adapter image two</figcaption>
+  </div>
+</div>
+
+Now pass the preprocessed masks to the pipelines `cross_attention_kwargs`.
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/ip_adapter.md:328]:
```suggestion
Now pass the preprocessed masks to `cross_attention_kwargs` in the pipeline call.
```

File: docs/source/en/using-diffusers/ip_adapter.md [hunk 8]
```diff
@@ -264,8 +275,77 @@ images = pipeline(
...
+generator = torch.Generator(device="cpu").manual_seed(0)
+num_images = 1
+
+image = pipeline(
+    prompt="2 girls",
+    ip_adapter_image=ip_images,
+    negative_prompt="monochrome, lowres, bad anatomy, worst quality, low quality",
+    num_inference_steps=20, num_images_per_prompt=num_images,
+    generator=generator, cross_attention_kwargs={"ip_adapter_masks": masks}
+).images[0]
```
Comment (sayakpaul) [inline docs/source/en/using-diffusers/ip_adapter.md:342]:
```suggestion
image = pipeline(
    prompt="2 girls",
    ip_adapter_image=ip_images,
    negative_prompt="monochrome, lowres, bad anatomy, worst quality, low quality",
    num_inference_steps=20, 
    num_images_per_prompt=num_images,
    generator=generator,
    cross_attention_kwargs={"ip_adapter_masks": masks}
).images[0]
```

Pythonic.

Comment (sayakpaul) [reply docs/source/en/using-diffusers/ip_adapter.md:342]: Does it also make sense to include results without the masks?

**Review by sayakpaul — COMMENTED** [review]
Very nice!

Left a couple of comments.

File: docs/source/en/using-diffusers/ip_adapter.md (hunk 3 above)
Comment (stevhliu) [reply docs/source/en/using-diffusers/ip_adapter.md:288]: done ✅

File: docs/source/en/api/loaders/ip_adapter.md (hunk 4 above)
Comment (stevhliu) [reply docs/source/en/api/loaders/ip_adapter.md:29]: Yeah they're here https://github.com/huggingface/diffusers/blob/687bc2772721af584d649129f8d2a28ca56a9ad8/src/diffusers/image_processor.py#L889

File: docs/source/en/using-diffusers/ip_adapter.md (hunk 8 above)
Comment (sayakpaul) [reply docs/source/en/using-diffusers/ip_adapter.md]: I think we need to pass the images with the exact same resolution. @stevhliu if it's possible, could you maybe create a separate issue thread for this with the code you tried and mention Yiyi and myself?

Comment (stevhliu) [reply docs/source/en/using-diffusers/ip_adapter.md]: The issue should be resolved now!

**Review by sayakpaul — APPROVED** [review]
Excellent work, Steven!

**Pull request merged**